storage/storage-interface/src/finality_view.rs

```diff
@@ -0,0 +1,159 @@
+use std::sync::RwLock;
+
+use aptos_crypto::HashValue;
+use aptos_types::{
+    aggregate_signature,
+    block_info::BlockInfo,
+    ledger_info::{LedgerInfo, LedgerInfoWithSignatures},
+    transaction::Version,
+};
+
+use crate::{AptosDbError, DbReader, Result};
+
+/// A wrapper over [`DbReader`], representing ledger at the end of a block
+/// as the latest ledger and its version as the latest transaction version.
+pub struct FinalityView<Db> {
+    reader: Db,
+    finalized_ledger_info: RwLock<Option<LedgerInfoWithSignatures>>,
+}
+
+impl<Db> FinalityView<Db> {
+    pub fn new(reader: Db) -> Self {
+        Self {
+            reader,
+            finalized_ledger_info: RwLock::new(None),
+        }
+    }
+}
+
+impl<Db: DbReader> FinalityView<Db> {
+    /// Updates the information on the latest finalized block at the specified height.
+    pub fn set_finalized_block_height(&self, height: u64) -> Result<()> {
+        let (_start_ver, end_ver, block_event) = self.reader.get_block_info_by_height(height)?;
+        let block_hash = block_event.hash()?;
+        let block_info = BlockInfo::new(
+            block_event.epoch(),
+            block_event.round(),
+            block_hash,
+            self.reader.get_accumulator_root_hash(end_ver)?,
+            end_ver,
+            block_event.proposed_time(),
+            None,
+        );
+        // Sanity checks: finalization should not be set on an empty database,
+        // the finality version should not exceed the latest committed.
+        match self.reader.get_latest_state_checkpoint_version()? {
+            None => return Err(AptosDbError::Other("no ledger states to finalize".into())),
+            Some(ver) => {
+                if end_ver > ver {
+                    return Err(AptosDbError::Other(format!(
+                        "finality version {end_ver} exceeds committed version {ver}"
+                    )));
+                }
+            },
+        }
+        let ledger_info = LedgerInfo::new(block_info, HashValue::zero()); // we don't use consensus_data_hash
+        let aggregate_signature = aggregate_signature::AggregateSignature::empty(); // we don't use
```

Comment:
Maybe provide an extended comment here.

Reply:
Explain why we don't use.

Reply:
This is one reason why, with my aptos-core hat on, I wanted to pass `LedgerInfoWithSignatures` as input to this API and synthesize it at the maptos-fin-view level where we know we can set these fields with trivial values.

Reply:
Thanks for picking on this. I think I have figure out a proper way to retrieve the ledger info; it's the latest one committed for the block's epoch.

Reply:
The idea did not work out without modifying more of `DbReader`, so we'll have to make do with the current method.
I have added a comment describing the Movement-specific construction of `LedgerInfoWithSignatures`.